var/spack/repos/builtin/packages/conduit/package.py

```diff
@@ -0,0 +1,289 @@
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+import socket
+import os
+from os.path import join as pjoin
+
+import llnl.util.tty as tty
+
+
+def cmake_cache_entry(name, value):
+    """
+    Helper that creates CMake cache entry strings used in
+    'host-config' files.
+    """
+    return 'set("%s" "%s" CACHE PATH "")\n\n' % (name, value)
+
+
+class Conduit(Package):
+    """Conduit is an open source project from Lawrence Livermore National
+    Laboratory that provides an intuitive model for describing hierarchical
+    scientific data in C++, C, Fortran, and Python. It is used for data
+    coupling between packages in-core, serialization, and I/O tasks."""
+
+    homepage = "http://software.llnl.gov/conduit"
+    url = "https://github.com/LLNL/conduit/archive/v0.2.0.tar.gz"
+
+    version('0.2.0', 'd595573dedf55514c11d7391092fd760')
+
+    version('github-master', git='https://github.com/LLNL/conduit.git')
+
+    ###########################################################################
+    # package variants
+    ###########################################################################
+
+    variant("shared", default=True, description="Build Conduit as shared libs")
+
+    variant("cmake", default=True,
+            description="Build CMake (if off, attempt to use cmake from PATH)")
+
+    # variants for python support
+    variant("python", default=True, description="Build Conduit Python support")
+
+    # variants for i/o
+    variant("mpi", default=True, description="Build Conduit MPI Support")
+    variant("hdf5", default=True, description="Build Conduit HDF5 support")
+    variant("silo", default=True, description="Build Conduit Silo support")
+
+    # variants for dev-tools (docs, etc)
+    variant("docs", default=True,   description="Build Conduit's docs")
+
+    ###########################################################################
+    # package dependencies
+    ###########################################################################
+
+    #######################
+    # CMake
+    #######################
+    depends_on("cmake@3.3.1", when="+cmake")
+
+    #######################
+    # Python
+    #######################
+    depends_on("python", when="+python")
+    depends_on("py-numpy~blas~lapack", when="+python")
+    depends_on("py-sphinx", when="+python+docs")
+    depends_on("doxygen", when="+docs")
+
+    #######################
+    # I/O Packages
+    #######################
+    depends_on("hdf5+shared~cxx~mpi~fortran", when="+shared")
+    depends_on("hdf5~shared~cxx~mpi~fortran", when="~shared")
+
+    depends_on("silo+shared~fortran", when="+shared")
+    depends_on("silo~shared~fortran", when="~shared")
```

Comment:
I agree with @adamjstewart, this should read:
```
depends_on('hdf5', when='+hdf5')
depends_on('silo', when='+silo')
```

1. If package A depends on package B (A->B), then `A+shared` will link just fine with `B+static`.  So there is no need to forward the `shared` variant.  We don't do it elsewhere.  Users who need to build static will need to set the `+static` (or ~shared; we're not yet quite sure which) on every package that can take it.  We're still considering good ways to do that; see #2380 #2594 #2492 #2165 #2644 

2. `packages.yaml` can be set at many levels: per-Spack, per-user, etc.  See: https://spack.readthedocs.io/en/latest/configuration.html  It's more than just Spack-wide.

> These selections are not only due to build issues.

Then they should definitely go in `packages.yaml`.